src/content-steering-controller.js

```diff
@@ -0,0 +1,317 @@
+ * https://datatracker.ietf.org/doc/draft-pantos-hls-rfc8216bis/ section 4.4.6.6.
+ * DASH: https://dashif.org/docs/DASH-IF-CTS-00XX-Content-Steering-Community-Review.pdf
+ *
+ * @param {Object} segmentLoader a reference to the mainSegmentLoader
+ */
+export default class ContentSteeringController extends videojs.EventTarget {
+  // pass a segment loader reference for throughput rate and xhr
+  constructor(segmentLoader) {
+    super();
+
+    this.currentPathway = null;
+    this.defaultPathway = null;
+    this.queryBeforeStart = null;
+    this.availablePathways_ = new Set();
+    // TODO: Implement exclusion.
+    this.excludedPathways_ = new Set();
+    this.steeringManifest = new SteeringManifest();
+    this.proxyServerUrl_ = null;
+    this.manifestType_ = null;
+    this.ttlTimeout_ = null;
+    this.request_ = null;
+    this.mainSegmentLoader_ = segmentLoader;
+    this.logger_ = logger('Content Steering');
+  }
+
+  /**
+   * Assigns the content steering tag properties to the steering controller
+   *
+   * @param {string} baseUrl the baseURL from the manifest for resolving the steering manifest url
+   * @param {Object} steeringTag the content steering tag from the main manifest
+   */
+  assignTagProperties(baseUrl, steeringTag) {
+    this.manifestType_ = steeringTag.serverUri ? 'HLS' : 'DASH';
+    // serverUri is HLS serverURL is DASH
+    const steeringUri = steeringTag.serverUri || steeringTag.serverURL;
+
+    if (!steeringUri) {
+      this.logger_(`steering manifest URL is ${steeringUri}, cannot request steering manifest.`);
+      this.trigger('error');
+      return;
+    }
+    // Content steering manifests can be encoded as a data URI. We can decode, parse and return early if that's the case.
+    if (steeringUri.startsWith('data:')) {
+      this.decodeDataUriManifest_(steeringUri.substring(steeringUri.indexOf(',') + 1));
+      return;
+    }
+    this.steeringManifest.reloadUri = resolveUrl(baseUrl, steeringUri);
+    // pathwayId is HLS defaultServiceLocation is DASH
+    this.defaultPathway = steeringTag.pathwayId || steeringTag.defaultServiceLocation;
+    // currently only DASH supports the following properties on <ContentSteering> tags.
+    if (this.manifestType_ === 'DASH') {
+      this.queryBeforeStart = steeringTag.queryBeforeStart || false;
+      this.proxyServerUrl_ = steeringTag.proxyServerURL;
+    }
+
+    // trigger a steering event if we have a pathway from the content steering tag.
+    // this tells VHS which segment pathway to start with.
+    if (this.defaultPathway) {
+      this.trigger('content-steering');
+    }
+  }
+
+  /**
+   * Requests the content steering manifest and parse the response.
+   */
+  requestSteeringManifest() {
+    // add parameters to the steering uri
+    const reloadUri = this.steeringManifest.reloadUri;
+    const uri = this.proxyServerUrl_ ? this.setProxyServerUrl_(reloadUri) : this.setSteeringParams_(reloadUri);
```

Comment:
This could be another TODO as I don't believe it is make or break, but are we accounting for params that are tied to the original MPD url for DASH? See Section 8.1 in https://dashif.org/docs/DASH-IF-CTS-00XX-Content-Steering-Community-Review.pdf

Reply:
So yeah I actually coded up support for this then deleted it after I realized it's tied to a specific MPD tag. 😅 
`<up:ExtUrlQueryInfo queryTemplate="token=$query:steeringToken$" useMPDUrlQuery="true"
includeInRequests="steering"/> </up:ExtUrlQueryInfo>`
I agree though, lets add at least a comment citing lack of support.